src/webots/nodes/WbTransform.cpp

```diff
@@ -52,6 +60,24 @@ WbTransform::~WbTransform() {
     wr_node_delete(WR_NODE(wrenNode()));
 }
 
+void WbTransform::reset(const QString &id) {
+  WbGroup::reset(id);
+  // note: for solids, the set of these parameters has to occur only if mJointParents.size() == 0 and it is handled in
+  // WbSolid::reset, otherwise it breaks the reset of hinge based joints
+  if (nodeType() != WB_NODE_TRACK_WHEEL && nodeType() != WB_NODE_SOLID) {
```

Comment:
What I meant was to use dynamic cast and not the `nodeType()` method, i.e.
```suggestion
  if (!dynamic_cast<const WbTrackWheel *>(this) && !dynamic_cast<const WbSolid *>(this)) {
```

Do you really want to skip only WbSolid nodes or any WbSolid-derived nodes (for example devices) that could also have parent or descendant joint nodes?

Reply:
Good point, doing so would alter the reset logic for those cases. As `SolidDevices` inherit the reset from `Solid` the change must apply to them as well. However can't use the cast for track wheel as it would introduce a circular dependency of `WbTransform`. Since nothing else depends on it (to my knowledge) it should be fine to leave that one as a nodeType check unless there's a better way.